src/main/java/com/hivemq/cli/commands/cli/TestBrokerCommand.java

```diff
@@ -0,0 +1,370 @@
+    private @NotNull Integer qosTries;
+
+    @CommandLine.Mixin
+    private AuthenticationOptions authenticationOptions = new AuthenticationOptions();
+
+    @CommandLine.Mixin
+    private SslOptions sslOptions = new SslOptions();
+
+    private @Nullable MqttClientSslConfig sslConfig;
+
+    //needed for pico cli - reflection code generation
+    private final DefaultCLIProperties defaultCLIProperties;
+
+    public TestBrokerCommand() {
+        this(null);
+    }
+
+    @Inject
+    public TestBrokerCommand(final @NotNull DefaultCLIProperties defaultCLIProperties) {
+        this.defaultCLIProperties = defaultCLIProperties;
+    }
+
+    @Override
+    public void run() {
+
+        // TinyLog configuration
+        Map<String, String> configurationMap = new HashMap<String, String>() {{
+            put("writer", "console");
+            put("writer.format", "{message}");
+            put("writer.level", "warn");
+        }};
+
+        Configuration.replace(configurationMap);
+
+        if (host == null) {
+            host = defaultCLIProperties.getHost();
+        }
+        if (port == null) {
+            port = defaultCLIProperties.getPort();
+        }
+
+        try {
+            sslConfig = sslOptions.buildSslConfig();
+        } catch (Exception e) {
+            Logger.error(e, "Could not build SSL configuration");
+        }
+
+        if (version != null) {
+            if (version == MqttVersion.MQTT_3_1_1) {
+                testMqtt3Features();
+            } else if (version == MqttVersion.MQTT_5_0) {
+                testMqtt5Features();
+            }
+        } else {
+            testMqtt3Features();
+            testMqtt5Features();
+        }
+    }
+
+    public void testMqtt5Features() {
+        final Mqtt5FeatureTester tester = new Mqtt5FeatureTester(
+                host,
+                port,
+                authenticationOptions.getUser(),
+                authenticationOptions.getPassword(),
+                sslConfig,
+                timeOut
+        );
+
+        boolean mqtt5Support = false;
+
+        // Test if MQTT5 is supported
+        System.out.print("MQTT 5: ");
+        final Mqtt5ConnAck connAck = tester.testConnect();
+        if (connAck == null) {
+            System.out.println("NO");
+        } else if (connAck.getReasonCode() == Mqtt5ConnAckReasonCode.SUCCESS) {
+            mqtt5Support = true;
+            System.out.println("OK");
+        } else {
+            System.out.println(connAck.getReasonCode().toString());
+        }
+
+        if (mqtt5Support) {
+
+            //*********************//
+            /* Connect Restrictions */
+            //*********************//
+
+            final Mqtt5ConnAckRestrictions restrictions = connAck.getRestrictions();
+
+            System.out.println("\t- Connect Restrictions: ");
```

Comment:
```suggestion
            System.out.println("\t- Connect restrictions: ");
```
be consistent with capitalization